test-framework/common/src/main/java/io/quarkus/test/common/DockerContainerLauncher.java

```diff
@@ -64,10 +64,14 @@ public void start() throws IOException {
         args.add("docker"); // TODO: determine this dynamically?
         args.add("run");
         args.add("--rm");
-        args.add("-p");
-        args.add(httpPort + ":" + httpPort);
-        args.add("-p");
-        args.add(httpsPort + ":" + httpsPort);
+        if ((httpPort == 0) || (httpsPort == 0)) {
+            args.add("--net=host");
```

Comment:
Sorry, I'm a little out of the context of this pull request but I will try to contribute.

Doubt

1. Is this random port generated by quarkus and then a container is started?
2. If I upload the container with the network = `--host` it will force the communication with other containers to be `--host`, too, right? (example access to postgresql)

Reply:
> Sorry, I'm a little out of the context of this pull request but I will try to contribute.
> 
> Doubt
> 
> 1. Is this random port generated by quarkus and then a container is started?

Yes, the port is determined by Quarkus when the application starts

> 2. If I upload the container with the network = `--host` it will force the communication with other containers to be `--host`, too, right? (example access to postgresq, testcontainers)

Not sure what exactly you mean here. Can you give a concrete example please?

Reply:
if my quarkus app starts in one container and has to communicate with another container they need it on the same network.

if I have, for example, a redis implementation and are using testcontainers, before my application goes up the testcontainer will start a redis container, right? at this point here the redis must also be on `--net=host` for the communication to work.

Reply:
That is indeed the case, however we don't really have much choice.

I guess we could theoretically pick a random port before starting the application under test and then map that port.

@gastaldi WDYT?

Reply:
Maybe a solution is to make `--net=host` be the default always, independently of the port chosen?  

Picking a random port before starting the application might work, but it is not a much reliable strategy.

Reply:
I updated the PR to have the test itself select the random ports

Reply:
I agree that we need something consistent, that's a good point that came out of this discussion.

What I did is update the PR to have the test itself select the random ports which I think is reasonable for now. 
In the future if we feel there is a good reason to resort to `--net=host`, we can certainly reevaluate

Reply:
thanks for your attention

Reply:
yw